python/lsst/pipe/tasks/imageDifference.py

```diff
@@ -348,6 +401,39 @@ def makeIdFactory(expId, expBits):
         """
         return afwTable.IdFactory.makeSource(expId, 64 - expBits)
 
+    def runQuantum(self, butlerQC: pipeBase.ButlerQuantumContext,
+                   inputRefs: pipeBase.InputQuantizedConnection,
+                   outputRefs: pipeBase.OutputQuantizedConnection):
+        """Pipelinetask gen3 entry point.
+
+        Parameters
+        ----------
+        expId : `int`
+            Exposure id.
+
+        expBits: `int`
+            Number of used bits in ``expId``.
+
+        Returns
+        -------
+        None
+        """
+        if self.getTemplate.config.coaddName == 'dcr':
+            # TODO DM-22952
+            raise NotImplementedError("TODO DM-22952: Dcr coadd templates are not yet supported in gen3.")
+
+        inputs = butlerQC.get(inputRefs)
+        expId, expBits = butlerQC.quantum.dataId.pack("visit_detector",
```

Comment:
This is missing an important log message that is in the Gen 2 version: `self.log.info("Processing %s" % (sensorRef.dataId))`. I think you can get this here with `self.log.info("Processing %s" % (inputs["exposure"].dataId))`